test/discr/grid_test.py

```diff
@@ -69,127 +69,102 @@ def test_init(as_midp):
+        TensorGrid(sorted1, with_nan, sorted2)
 
     with pytest.raises(ValueError):
-        TensorGrid(sorted1, with_inf, sorted2, as_midp=as_midp)
+        TensorGrid(sorted1, with_inf, sorted2)
 
     with pytest.raises(ValueError):
-        TensorGrid(sorted1, empty, sorted2, as_midp=as_midp)
+        TensorGrid(sorted1, empty, sorted2)
 
 
-def test_ndim(as_midp):
-    vec1 = np.arange(2, 6)
-    vec2 = np.arange(-4, 5, 2)
+def test_tensorgrid_ndim():
+    vec1 = np.array([2, 3, 4, 5])
+    vec2 = np.array([-4, -2, 0, 2, 4])
     vec3 = np.linspace(-2, 2, 50)
 
-    grid1 = TensorGrid(vec1, as_midp=as_midp)
-    grid2 = TensorGrid(vec1, vec2, vec3, as_midp=as_midp)
+    grid1 = TensorGrid(vec1)
+    grid2 = TensorGrid(vec1, vec2, vec3)
 
     assert grid1.ndim == 1
     assert grid2.ndim == 3
 
 
-def test_shape(as_midp):
-    vec1 = np.arange(2, 6)
+def test_tensorgrid_shape():
+    vec1 = np.array([2, 3, 4, 5])
     vec2 = np.linspace(-2, 2, 50)
     scalar = 0.5
 
-    grid1 = TensorGrid(vec1, as_midp=as_midp)
-    grid2 = TensorGrid(vec1, vec2, as_midp=as_midp)
-    grid3 = TensorGrid(scalar, vec2, as_midp=as_midp)
+    grid1 = TensorGrid(vec1)
+    grid2 = TensorGrid(vec1, vec2)
+    grid3 = TensorGrid(scalar, vec2)
 
     assert grid1.shape == (4,)
     assert grid2.shape == (4, 50)
     assert grid3.shape == (1, 50)
 
 
-def test_size(as_midp):
-    vec1 = np.arange(2, 6)
+def test_tensorgrid_size():
+    vec1 = np.array([2, 3, 4, 5])
     vec2 = np.linspace(-2, 2, 50)
     scalar = 0.5
 
-    grid1 = TensorGrid(vec1, as_midp=as_midp)
-    grid2 = TensorGrid(vec1, vec2, as_midp=as_midp)
-    grid3 = TensorGrid(scalar, vec2, as_midp=as_midp)
+    grid1 = TensorGrid(vec1)
+    grid2 = TensorGrid(vec1, vec2)
+    grid3 = TensorGrid(scalar, vec2)
 
     assert grid1.size == 4
     assert grid2.size == 200
     assert grid3.size == 50
 
 
-def test_minpt_maxpt(as_midp):
-    vec1 = np.arange(2, 6)
-    vec2 = np.arange(-4, 5, 2)
-    vec3 = np.arange(-1, 1)
+def test_tensorgrid_minpt_maxpt():
+    vec1 = np.array([2, 3, 4, 5])
+    vec2 = np.array([-4, -2, 0, 2, 4])
+    vec3 = np.array([-1, 0])
     scalar = 0.5
 
-    grid = TensorGrid(vec1, vec2, vec3, as_midp=as_midp)
+    grid = TensorGrid(vec1, vec2, vec3)
     assert all_equal(grid.min_pt, (2, -4, -1))
     assert all_equal(grid.max_pt, (5, 4, 0))
 
-    grid = TensorGrid(vec1, scalar, vec2, scalar, as_midp=as_midp)
+    grid = TensorGrid(vec1, scalar, vec2, scalar)
     assert all_equal(grid.min_pt, (2, 0.5, -4, 0.5))
     assert all_equal(grid.max_pt, (5, 0.5, 4, 0.5))
 
 
-def test_element(as_midp):
-    vec1 = np.arange(2, 6)
-    vec2 = np.arange(-4, 5, 2)
+def test_tensorgrid_element():
+    vec1 = np.array([2, 3, 4, 5])
+    vec2 = np.array([-4, -2, 0, 2, 4])
 
-    grid = TensorGrid(vec1, vec2, as_midp=as_midp)
+    grid = TensorGrid(vec1, vec2)
     some_pt = grid.element()
     assert some_pt in grid
 
 
-def test_min_max():
```

Comment:
I can add back a simple version. But don't doctests also contribute to coverage?